### PR TITLE
Let the FlutterAppDelegate's messenger be customizable if the rootViewController isn't a FlutterViewController

### PR DESCRIPTION
**Pull request by xster**
https://github.com/flutter/flutter/issues/11349

### CODE REVIEW

**Comment by Hixie** [comment]
Can we test this?

File: shell/platform/darwin/ios/framework/Headers/FlutterAppDelegate.h [hunk 1]
```diff
@@ -29,6 +29,12 @@ FLUTTER_EXPORT
 
 @property(strong, nonatomic) UIWindow* window;
 
+// Can be overriden by subclasses to provide a custom FlutterBinaryMessenger,
+// typically a FlutterViewController, for plugin interop.
+//
+// Defaults to window's rootViewController.
+- (NSObject<FlutterBinaryMessenger>*) pluginMessenger;
```
Comment (mravn-google) [inline shell/platform/darwin/ios/framework/Headers/FlutterAppDelegate.h]: Maybe rename to `binaryMessenger` or `messenger`?

**Comment by Hixie** [comment]
What is the status of this PR?

File: shell/platform/darwin/ios/framework/Headers/FlutterAppDelegate.h (hunk 1 above)
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterAppDelegate.h]: Sure, makes sense

**Comment by xster** [comment]
We'll need first https://github.com/flutter/flutter/issues/11483 to add tests

**Pull request merged**